### PR TITLE
[AMDGPU] Do not print `kernel-resource-usage` information on non-kernels

### PR DESCRIPTION
**Pull request by jhuber6**
Summary:
This pass is used to get helpful information about the kernel resources
without needing to insepct the binary. However, it currently prints on
every function. These values will always be zero, so it's just spam on
the terminal, at best an indication that a function wasn't internalized
/ optimized out. This patch makes it only print for kernels to make it
more useful in practice.

### CODE REVIEW

File: llvm/lib/Target/AMDGPU/AMDGPUAsmPrinter.cpp [hunk 1]
```diff
@@ -1487,6 +1487,10 @@ void AMDGPUAsmPrinter::emitResourceUsageRemarks(
   if (!Ctx.getDiagHandlerPtr()->isAnalysisRemarkEnabled(Name))
     return;
 
+  // Currently non-kernel functions have no resources to emit.
+  if (MF.getFunction().getCallingConv() != CallingConv::AMDGPU_KERNEL)
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/AMDGPUAsmPrinter.cpp:1491]: This ignores graphics. Should use isEntryFunctionCC instead

Comment (jhuber6) [inline llvm/lib/Target/AMDGPU/AMDGPUAsmPrinter.cpp:1491]:
```suggestion
  if (!isEntryFunctionCC(MF.getFunction().getCallingConv()))
```

File: llvm/test/CodeGen/AMDGPU/resource-optimization-remarks.ll [hunk 2]
```diff
@@ -144,15 +135,7 @@ define amdgpu_kernel void @empty_kernel() !dbg !7 {
...
-; STDERR: remark: foo.cl:52:0: Function Name: empty_func
-; STDERR-NEXT: remark: foo.cl:52:0:     SGPRs: 0
-; STDERR-NEXT: remark: foo.cl:52:0:     VGPRs: 0
-; STDERR-NEXT: remark: foo.cl:52:0:     AGPRs: 0
-; STDERR-NEXT: remark: foo.cl:52:0:     ScratchSize [bytes/lane]: 0
-; STDERR-NEXT: remark: foo.cl:52:0:     Dynamic Stack: False
-; STDERR-NEXT: remark: foo.cl:52:0:     Occupancy [waves/SIMD]: 0
-; STDERR-NEXT: remark: foo.cl:52:0:     SGPRs Spill: 0
-; STDERR-NEXT: remark: foo.cl:52:0:     VGPRs Spill: 0
+; STDERR-NOT: remark: foo.cl:52:0: Function Name: empty_func
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/resource-optimization-remarks.ll:138]: Not checks should be as simple as possible. Just -NOT: remark. Or add -implicit-check-not=remark

Comment (jhuber6) [inline llvm/test/CodeGen/AMDGPU/resource-optimization-remarks.ll:138]:
```suggestion
; STDERR-NOT: empty_func
```

File: llvm/test/CodeGen/AMDGPU/resource-optimization-remarks.ll [hunk 3]
```diff
@@ -113,16 +113,7 @@ define amdgpu_kernel void @test_kernel() !dbg !3 {
...
-; STDERR-NEXT: remark: foo.cl:42:0:     SGPRs: 0
-; STDERR-NEXT: remark: foo.cl:42:0:     VGPRs: 0
-; STDERR-NEXT: remark: foo.cl:42:0:     AGPRs: 0
-; STDERR-NEXT: remark: foo.cl:42:0:     ScratchSize [bytes/lane]: 0
-; STDERR-NEXT: remark: foo.cl:42:0:     Dynamic Stack: False
-; STDERR-NEXT: remark: foo.cl:42:0:     Occupancy [waves/SIMD]: 0
-; STDERR-NEXT: remark: foo.cl:42:0:     SGPRs Spill: 0
-; STDERR-NEXT: remark: foo.cl:42:0:     VGPRs Spill: 0
-; STDERR-NOT: LDS Size
+; STDERR-NOT: remark: foo.cl:42:0: Function Name: test_func
```
Comment (jhuber6) [inline llvm/test/CodeGen/AMDGPU/resource-optimization-remarks.ll:116]:
```suggestion
; STDERR-NOT: test_func
```

**Review by arsenm — APPROVED** [review]

**Pull request merged**